hbase-server/src/main/java/org/apache/hadoop/hbase/master/TableStateManager.java

```diff
@@ -53,8 +52,20 @@
 // TODO: Make this a guava Service
 @InterfaceAudience.Private
 public class TableStateManager {
-
   private static final Logger LOG = LoggerFactory.getLogger(TableStateManager.class);
+
+  /**
+   * All table state is kept in hbase:meta except that of hbase:meta itself.
+   * hbase:meta state is kept here locally in this in-memory variable. State
+   * for hbase:meta is not persistent. If this process dies, the hbase:meta
+   * state reverts to enabled. State is used so we can edit hbase:meta as we
+   * would any other table by disabling, altering, and then re-enabling. If this
+   * process dies in the midst of an edit, the table reverts to enabled. Schema
+   * is read from the filesystem. It is changed atomically so if we die midway
+   * through an edit we should be good.
+   */
+  private TableState.State metaTableState = TableState.State.ENABLED;
```

Comment:
Why not store it on zk? Just like the location of meta table...

Reply:
If the Master dies during enable/disable, I want it to default to being online when new Master comes online. Lots of new conditions if hbase:meta is offline.